xrpl4j-crypto-parent/xrpl4j-crypto-core/src/main/java/org/xrpl/xrpl4j/crypto/core/keys/PrivateKey.java

```diff
@@ -0,0 +1,113 @@
+package org.xrpl.xrpl4j.crypto.core.keys;
+
+import com.google.common.hash.HashCode;
+import com.google.common.hash.Hashing;
+import org.xrpl.xrpl4j.codec.addresses.UnsignedByte;
+import org.xrpl.xrpl4j.codec.addresses.UnsignedByteArray;
+import org.xrpl.xrpl4j.codec.addresses.VersionType;
+
+import java.util.Objects;
+
+/**
+ * A typed instance of an XRPL private-key.
+ */
+public class PrivateKey implements PrivateKeyable, javax.security.auth.Destroyable {
+
+  /**
+   * Keys generated from the secp256k1 curve have 33 bytes in XRP Ledger. However, keys derived from the ed25519 curve
+   * have only 32 bytes, and so get prefixed with this HEX value so that all keys in the ledger are 33 bytes.
+   */
+  public static final UnsignedByte PREFIX = UnsignedByte.of(0xED);
+
+  private final UnsignedByteArray value;
+  private final HashCode sha256HashCode;
```

Comment:
Is there a specific reason to compute the SHA256 of the key? I don't see `sha256()` used anywhere. Not a huge deal, but could be a performance hit if you're constructing a lot of `PrivateKey`s

Reply:
Yes, let's remove this.